Monika After Story/game/zz_backgrounds.rpy

```diff
@@ -0,0 +1,518 @@
+
+        def __ne__(self, other):
+            result = self.__eq__(other)
+            if result is NotImplemented:
+                return result
+            return not result
+
+
+        def entry(self, old_background):
+            """
+            Run the entry programming point
+            """
+            if self.entry_pp is not None:
+                self.entry_pp(old_background)
+
+
+        def exit(self, new_background):
+            """
+            Run the exit programming point
+            """
+            if self.exit_pp is not None:
+                self.exit_pp(new_background)
+
+
+        def fromTuple(self, data_tuple):
+            """
+            Loads data from tuple
+
+            IN:
+                data_tuple - tuple of the following format:
+                    [0]: unlocked property
+            """
+            self.unlocked = data_tuple[0]
+
+
+        def toTuple(self):
+            """
+            Converts this MASWeather object into a tuple
+
+            RETURNS: tuple of the following format:
+                [0]: unlocked property
+            """
+            return (self.unlocked,)
+
+        def getDayRoom(self, weather=None):
+            """
+            Returns the day masks to use given the conditions/availablity of present assets
+            """
+            if weather is None:
+                weather = store.mas_current_weather
+
+            if weather.precip_type == mas_weather.PRECIP_TYPE_RAIN:
+                return self.image_rain_day
+            elif weather.precip_type == mas_weather.PRECIP_TYPE_OVERCAST:
+                return self.image_overcast_day
+            elif weather.precip_type == mas_weather.PRECIP_TYPE_SNOW:
+                return self.image_snow_day
+            return self.image_day
+
+        def getNightRoom(self, weather=None):
+            """
+            Returns the night masks to use given the conditions/availablity of present assets
+            """
+            if weather is None:
+                weather = store.mas_current_weather
+
+            if weather.precip_type == mas_weather.PRECIP_TYPE_RAIN:
+                return self.image_rain_night
+            elif weather.precip_type == mas_weather.PRECIP_TYPE_OVERCAST:
+                return self.image_overcast_night
+            elif weather.precip_type == mas_weather.PRECIP_TYPE_SNOW:
+                return self.image_snow_night
+            return self.image_night
+
+        def getRoomForTime(self, weather=None):
+            """
+            Gets the room for the current time
+
+            IN:
+                weather - get the room bg for the time and weather
+                (Default: current weather)
+            """
+            if weather is None:
+                weather = store.mas_current_weather
+            if store.mas_isMorning():
+                return self.getDayRoom(weather)
+            return self.getNightRoom(weather)
+
+        def isChangingRoom(self, old_weather, new_weather):
```

Comment:
can you add documentation about this function? particularly what it means for a BG object to be "changing room"